### PR TITLE
publish-oc-binary.sh: fix `and` operator

### PR DESCRIPTION
**Pull request by bbguimaraes**

### CODE REVIEW

File: build-scripts/rcm-guest/publish-oc-binary.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@ extract() {
     mkdir macosx windows
     for arch in x86_64 ${ARCH}; do
         rpm=$(echo "$(rpm_name "${arch}")"*)
-        [ "${arch}" != x86_64 && ! -e "${rpm}" ] && continue
+        [ "${arch}" != x86_64 -a ! -e "${rpm}" ] && continue
```
Comment (dobbymoodge) [inline build-scripts/rcm-guest/publish-oc-binary.sh]: `[[ "${arch}" != x86_64 && ! -e "${rpm}" ]] && continue` ?

Comment (dobbymoodge) [reply build-scripts/rcm-guest/publish-oc-binary.sh]: @bbguimaraes or you could do `[ "${arch}" != x86_64 ] && [ ! -e "${rpm}" ] && continue` but I wouldn't :D

Comment (dobbymoodge) [reply build-scripts/rcm-guest/publish-oc-binary.sh]: @bbguimaraes Actually, the modeline is `#!/bin/sh`, so your fix is most proper, though I would quote both sides of the string comparison.

File: build-scripts/rcm-guest/publish-oc-binary.sh [hunk 2]
```diff
@@ -12,7 +12,7 @@ extract() {
...
         mkdir "${arch}"
         if [ "${arch}" != x86_64 ]; then
```
Comment (dobbymoodge) [inline build-scripts/rcm-guest/publish-oc-binary.sh]: Quote both sides here, too:
`if [ "${arch}" != "x86_64" ]; then`

**Comment by stevekuznetsov** [comment]
@juanvallejo @bbguimaraes why `#!/bin/sh`? If there is a good reason (or just in general, _always_ use ShellCheck:

```
$ shellcheck build-scripts/rcm-guest/publish-oc-binary.sh
 
Line 5:
    printf "${RPM}" "${arch}"
           ^-- SC2059: Don't use variables in the printf format string. Use printf "..%s.." "$foo".
 
Line 6:
    [ "${arch}" == x86_64 ] && printf %s -redistributable
                ^-- SC2039: In POSIX sh, == in place of = is undefined.
 
Line 11:
    local arch rpm
    ^-- SC2039: In POSIX sh, 'local' is undefined.
 
Line 15:
        [ "${arch}" != x86_64 && ! -e "${rpm}" ] && continue
                              ^-- SC2107: Instead of [ a && b ], use [ a ] && [ b ].
 
Line 18:
            rpm2cpio ${rpm}* | cpio -idm --quiet ./usr/bin/oc
                     ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 21:
            rpm2cpio ${rpm}* \
                     ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 23:
                    ./usr/share/atomic-openshift/{linux,macosx}/oc \
                                                 ^-- SC2039: In POSIX sh, brace expansion is undefined.
 
Line 33:
    local dir
    ^-- SC2039: In POSIX sh, 'local' is undefined.
 
Line 49:
trap "rm -rf '${TMPDIR}'" EXIT INT TERM
              ^-- SC2064: Use single quotes, otherwise this expands now rather than when signalled.
```

**Comment by bbguimaraes** [comment]
`#!/bin/sh` was used (or kept, rather) because the script previously did not
use any bash features.  I should have changed it when I added them.

About `shellcheck`'s output:

    Line 5:
        printf "${RPM}" "${arch}"
               ^-- SC2059: Don't use variables in the printf format string. Use printf "..%s.." "$foo".

Intentional, `RPM` contains the format string and is entirely under the
program's control.

    Line 49:
    trap "rm -rf '${TMPDIR}'" EXIT INT TERM
                  ^-- SC2064: Use single quotes, otherwise this expands now rather than when signalled.

Intentional.

    Line 11:
        local arch rpm
        ^-- SC2039: In POSIX sh, 'local' is undefined.

    Line 23:
                        ./usr/share/atomic-openshift/{linux,macosx}/oc \
                                                     ^-- SC2039: In POSIX sh, brace expansion is undefined.

    Line 33:
        local dir
        ^-- SC2039: In POSIX sh, 'local' is undefined.

These are the reason the script should use bash.

    Line 15:
            [ "${arch}" != x86_64 && ! -e "${rpm}" ] && continue
                                  ^-- SC2107: Instead of [ a && b ], use [ a ] && [ b ].

Actual bug.

    Line 6:
        [ "${arch}" == x86_64 ] && printf %s -redistributable
                    ^-- SC2039: In POSIX sh, == in place of = is undefined.

Ok, I'm genuinely surprised by this.

    Line 18:
                rpm2cpio ${rpm}* | cpio -idm --quiet ./usr/bin/oc
                         ^-- SC2086: Double quote to prevent globbing and word splitting.

    Line 21:
                rpm2cpio ${rpm}* \
                         ^-- SC2086: Double quote to prevent globbing and word splitting.

Not really a problem, given the inputs, but should be fixed.

**Comment by stevekuznetsov** [comment]
Yeah not saying you need to always address every output of ShellCheck, but there are almost always tangible changes that need to be addressed.

**Pull request merged**